Modules/Server/Sources/PocketCastsServer/Public/ServerPodcastManager.swift

```diff
@@ -40,20 +41,38 @@ public class ServerPodcastManager: NSObject {
     public func addFromUuidWithRetries(podcastUuid: String, subscribe: Bool, tries: UInt = 0, completion: ((Bool) -> Void)?) {
         var pollbackCounter = tries
         addFromUuid(podcastUuid: podcastUuid, subscribe: subscribe) { [weak self] success in
+            guard let self else {
+                return
+            }
+
             if success {
                 completion?(success)
                 return
             }
+
             pollbackCounter += 1
-            if pollbackCounter < 3 {
-                Thread.sleep(forTimeInterval: pow(2, Double(pollbackCounter)))
-                self?.addFromUuidWithRetries(podcastUuid: podcastUuid, subscribe: subscribe, tries: pollbackCounter, completion: completion)
+            if pollbackCounter < 8 {
+                Thread.sleep(forTimeInterval: pollBackoffTime(pollCount: pollbackCounter - 1))
+                addFromUuidWithRetries(podcastUuid: podcastUuid, subscribe: subscribe, tries: pollbackCounter, completion: completion)
                 return
             }
             completion?(false)
         }
     }
 
+    private func pollBackoffTime(pollCount: UInt) -> TimeInterval {
```

Comment:
Suggestion for a future PR: centralize these pollBackoffTime method in a central place, we have know three three similar implementations of this :)

Reply:
Done in ef491fff3908fdafee754ccb108e557c240a1c0d